www-reputation-credentials-protocol.md

```diff
@@ -0,0 +1,265 @@
+# Mitigating Channel Jamming with One-Shot HTLC Forward Reputation Credentials
+
+Authors: Antoine Riard <btc@ariard.me>
+
+Created: 2022-11-15
+
+# Problem Statement
+
+The channel jamming issue has been one of the longuest standing and unsolved issue affecting the
+Lightning Network. Not only attack has been demonstrated in practice, but also theoritically
+[studied](https://jamming-dev.github.io/book/): "A channel jamming is when a malicious entity blocks up liquidity in the LN,
+by making a payment to themselves, via third party channels and then never revealing the secret, such
+that the payment never completes".
+
+The channel jamming root cause comes from the following situation. A routed HTLC across multiple Lightning
+hops is a chain of contracts among trust-minimized counterparties. The settlement of this chain of contracts
+is uniform in its outcome (success/failure) across all the intermediary counterparties, however such counterparties
+cannot influence the outcome, once the chain of contract is established. As setting up this chain of contracts
+consume scarce resources (i.e the channel liquidity timevalue), there is a counterparty risk without guarantee
+there is an adequate compensation. Indeed, the routing fees are only paid in case of success.
+
+# Solution introduction
+
+This counterparty risk can be solved straightforwardly by the introduction of a premimum fee carried on by
+the HTLC sender. This jamming solution known as ["upfront fees"](https://github.com/t-bast/lightning-docs/blob/master/spam-prevention.md)
+in the Lightning community, aims to a fair distribution of the HTLC forward risk among all the chain of
+contracts counterparties. While there is a concern this risk distribution doesn't capture well the
+shenanigans of a HTLC forward (e.g offline routing hop, spontaneously congestioned channel), there is a
+major drawback as it introduces a permanent overhead fee for the network participants.
```

Comment:
From what I understand these reputation tokens are being sent as a form of upfront/unconditional monetary payment from the sender to each node in a route. Is this right?

Considering that, aside from these concerns you cite here, is there anything that makes paying these upfront fees with sats inherently harder or worse than paying with reputation tokens?

For example, in a route `A->B->C->D` is there something that prevents A from sending some extra msats unconditionally to B and C?

(I couldn't find anything that makes that a no-go on https://github.com/t-bast/lightning-docs/blob/master/spam-prevention.md)

Reply:
This is correct these reputation tokens are sent as a form of upfront/unconditional monetary payments. As noted on the mailing list, they might in fact belong more to the upfront/unconditional family of schemes rather than "pure reputation".

The major difference, minimizing the permanent overhead fee, is to stake the credentials across the HTLC sends. In the optimistic case (no HTLC failure), you only pay for few credentials once, then you should get more of them as your HTLC sends succeed, rather than spending a small upfront fee each time.

There is nothing that prevents A from sending extra msats unconditionally to B and C. The reason of the dissemination phase is for privacy reasons, otherwise you're disclosing to the routing hop this is the first time you're requesting a HTLC forward.

Reply:
In a way, things look simpler without a dissemination phase. For each hop you either attach a valid token, or pay an unconditional upfront fee. Either way, you'll get a new free token back for the next payment attempt - assuming no misbehavior.

On the other hand this does require different changes to allow these unconditional fees.

Reply:
All though yes, this is probably the same as always requiring an upfront fee and lowering the routing fee slightly. In both cases, you pay a little bit for a failure and more for a success.

Reply:
The main motivation of introducing "reputational" credentials is to stake/accumulate them across time, therefore making your future HTLC sends cheaper, optismistically without introducing traffic correlation between your HTLC forward requests. Instead of using a credential, a discount in function of actual hold time could be applied on the success fee to cover next payment's upfront fee, though I think you're limited to accumulate discounts across a period. For long-term held payments, this is an issue as the upfront fees should probably scale up to cover the corresponding routing hop risk.

There is also another advantage of introducing credentials, if they can be transferred, a LSP could pay for the upfront fees/unconditional fees of its spokes, therefore covering the jamming risk they represent for routing hops in the network.